src/wp-admin/includes/image.php

```diff
@@ -629,6 +633,11 @@ function wp_generate_attachment_metadata( $attachment_id, $file ) {
 	// Remove the blob of binary data from the array.
 	unset( $metadata['image']['data'] );
 
+	// Capture file size for cases where it has not been captured yet, such as PDFs.
+	if ( ! isset( $metadata['filesize'] ) && file_exists( $file ) ) {
```

Comment:
Just a super edge case where this won't work, when the `wp_filesize` has a filter that returns `null` so this value would be set but to `null` in that scenario it makes more sense to use `array_key_exists` instead.

```suggestion
	if ( ! array_key_exists( 'filesize', $metadata ) && file_exists( $file ) ) {
```

Reply:
IMO that works well as is, the `array_key_exists` would be overkill especially since `wp_filesize()` should never return `null` and if somebody uses the filter like that, they're doing it wrong.

With your above comment to enforce it to be an integer (which I agree with), this would become unnecessary anyway.

Reply:
I agree, I don't see the issue with the way it is.